### PR TITLE
Add Test Env Require Approval Action

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Add Test Env Require Approval Action.

Every time when a PR get created:
1. If PR author is not part of `.github/CODEOWNERS`, assign `ml-commons-cicd-env-required-approval`, pending for review by the assigned maintainers of the env.
2. If PR author is a maintainer, then assign `ml-commons-cicd-env`, no need approval.
3. If it is a merge which resulted in a push, `ml-commons-cicd-env` assigned, no need approval.

Example waiting for approval by maintainer: https://github.com/opensearch-project/ml-commons/pull/3004.

Thanks.

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/ml-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Adding @ylwu-amzn @dhrubo-os to take a look.

Note you have to merge this PR before it take effect as it is Pull_Request_Target event.

Thanks.

File: .github/workflows/require-approval.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          ref: ${{ github.event.pull_request.base.sha }}
+      - name: Get CodeOwner List
+        id: step-is-require-approval
+        run: |
+          github_event=${{ github.event_name }}
+          if [[ "$github_event" = "push" ]]; then
+            echo "Push event does not need approval"
+            echo "is-require-approval=ml-commons-cicd-env" >> $GITHUB_OUTPUT
+          else
+            approvers=$(cat .github/CODEOWNERS | grep @ | tr -d '* ' | sed 's/@/,/g' | sed 's/,//1')
```
Comment (dhrubo-os) [inline .github/workflows/require-approval.yml:27]: So currently, it's maintainers vs non-maintainers, right?

not maintainers + organization vs non-maintainers?

File: .github/workflows/require-approval.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            author=${{ github.event.pull_request.user.login }}
+            if [[ "$approvers" =~ "$author" ]]; then
+              echo "$authoer is in the approval list"
```
Comment (dhrubo-os) [inline .github/workflows/require-approval.yml:30]: [nit] author

File: .github/workflows/require-approval.yml (hunk 1 above)
Comment (peterzhuamazon) [reply .github/workflows/require-approval.yml:27]: Yes, it is easier to get maintainers list, we will start from here and improve.

File: .github/workflows/require-approval.yml (hunk 2 above)
Comment (peterzhuamazon) [reply .github/workflows/require-approval.yml]: Updated, thanks.

**Review by dhrubo-os — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]
Thanks Peter for enhancing this

**Pull request merged**